scraper/src/meilisearch_helper.py

```diff
@@ -101,11 +102,16 @@ class MeiliSearchHelper:
         'acceptNewFields': False
     }
 
-    def __init__(self, host_url, api_key, index_uid):
+    def __init__(self, host_url, api_key, index_uid, custom_settings):
         self.meilisearch_client = meilisearch.Client(host_url, api_key)
         self.__delete_and_create_index(index_uid)
         self.meilisearch_index = self.__delete_and_create_index(index_uid)
-        self.meilisearch_index.update_settings(MeiliSearchHelper.SETTINGS)
+        settings = {**MeiliSearchHelper.SETTINGS, **custom_settings}
+        try:
+            self.meilisearch_index.update_settings(settings)
+        except Exception:
+            print("MeiliSearchApiError: Settings are badly formatted. ")
+            sys.exit()
```

Comment:
No need to try/catch the Exception :)
For the moment the python package will raise an exception, even if it's not the best exception in the world 😉 The python package will be upgraded soon.
[Algolia does not try/catch at this point](https://github.com/algolia/docsearch-scraper/blob/5ce1d095bb76b0aca87d47dc8f7e045d2cf30f44/scraper/src/algolia_helper.py#L18), keep it simple 🙂

Reply:
If I remove the sys.exit() the scrape will continue ignoring the fact that the settings were not taken into account. Do we want that?

Reply:
@bidoubiwa No, it will not continue: raising an exception stops the process.

Reply:
You'r right. I thought I tried it and it did not raise anything but I probably made an error during my test. 
Removing it!